todo_app/tests/test_integration.py

```diff
@@ -0,0 +1,48 @@
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+import requests, os, pytest
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+    test_app = app.create_app()
+    with test_app.test_client() as client:
+        yield client
+
+def test_index_page(monkeypatch, client):
+    monkeypatch.setattr(requests, 'request', get_lists_stub)
+    response = client.get('/')
+    assert response.status_code == 200
+    assert "Test card" in response.data.decode()
+
+def test_remove(monkeypatch, client):
```

Comment:
This is a slightly odd test because the structure definitely feels reasonable but I'm not convinced it's testing what we want it to!

If we think about it, the way your stub is set up is that the result of an appropriate `delete` request will always return an empty list itself but future get requests to the board will see the card reappear. This test is coincidentally passing because the response returned from the POST request is a redirect, which doesn't mention any cards ever (take a look by printing the response) 

If we add a follow_redirects flag (like below) so that we load the index page after the deletion, then we see that the card does still appear!
```python
response = client.post("/remove/1A", follow_redirects=True)
```

If we were writing a test like this, I'd be tempted to have a single fake_response_data object that your stub returns, which can be _modified_ by the delete request, so that we do actually change the cards that are returned when the index is next loaded

Reply:
(Regardless of the above, the test clearly shows you know how to use all the relevant pieces, so nice work there! 😄 )